feat(v2): support markup within string

MMMalik opened the pull request:
The goal of this PR was to add support for setting `initData` as string with markup. For now we support only JSX.

JSX (supported now):
```js
<CKEditor initData={<p>Hello <strong>world</strong>!</p>} />
```

Strings with markup (to be supported thanks to this PR):
```js
<CKEditor initData="<p>Hello <strong>world</strong>!</p>" />

```

I see few ways to achieve an interface like that.

1. Set `initData` with `editor.setData` if it's string with markup. Use `children` if `initData` is JSX. With this approach there are two different but complementary ways of setting initial data. It's a bit complex solution.

2. Transform `initData` to string with [renderToStaticMarkup](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup) if it's JSX and then use `editor.setData`. This solution requires us to use `react-dom/server` in the browser which has disadvantages such as bigger library size and more complex dependencies.

3. Use `dangerouslySetInnerHTML` to set initial data if it's string with markup. Use `children` if `initData` is JSX. This is a potential security vulnerability because content would be rendered to DOM with `dangerouslySetInnerHTML`.

I decided to use approach 1. It's not ideal but I believe it's better than other solutions. I have purposefully added `initContent` prop to `useCKEditor` rather than naming it `initData`. It's conceptually different than prop in `CKEditor` component and it supports `string` only (no JSX).

Alternatively we can give up on supporting either JSX or strings with markup which would lead to simpler code. I believe though that there are valid use cases for both. JSX is a language of React, so developers might re-use their components within `initData`. Developers might also want to use strings with markup if it comes form external sources.

Closes #228.

Comandeer commented on src/useCKEditor.ts:
```diff
@@ -173,6 +179,25 @@ function useCKEditor<EditorEvent extends string>( {
+								 */
+								noSnapshot: true,
+
+								/**
+								 * Resets undo stack.
+								 */
+								callback: () => {
+									editor.resetUndo();
+								}
```
The callback can be probably safely omitted as it's the first `setData` and there's already `noSnapshot` option set to `true`. Due to that undo stack should be empty.

MMMalik replied:
Indeed, `callback` option is unnecessary. I've removed it and observed no issues.

Comandeer reviewed (changes requested):
The solution looks and works well 👍🏻 I have only one small comment (see below).

Additionally there isn't any sample using the new way of setting initial data. We probably should add one.

MMMalik commented:
Thanks @Comandeer, I've addressed your inline comment.

> Additionally there isn't any sample using the new way of setting initial data. We probably should add one.

I'm not sure about adding brand new sample just for the sake of testing new way of setting `initData`. Instead, I've used the new way in `basic` sample and added a comment about complementary possibilities. WDYT?

Comandeer commented:
> I'm not sure about adding brand new sample just for the sake of testing new way of setting `initData`. Instead, I've used the new way in `basic` sample and added a comment about complementary possibilities. WDYT?

SGTM :+1:

Comandeer commented:
Rebased onto latest `major`.

Comandeer reviewed (changes requested):
It seems that after all callback was needed… I've discovered a very weird bug with undo:

1. Open basic sample.
2. Write any character in the editor.
3. Press Undo.

The undo button should be disabled after one press, but it's still active and another press changes the content of the editor to "undefined".

Sorry for the trouble 😞

MMMalik commented:
Hi @Comandeer,

indeed, it looks like both options must be passed to `setData` in order to set initial data correctly:

- `noSnapshot` prevents undo icon flickering
- `editor.resetUndo` clears undo stack correctly

Please check now.

Comandeer reviewed (approved):
LGTM! 🎉
